app/src/main/java/com/example/henzoshimada/feeltrip/loginActivity.java

```diff
@@ -18,6 +20,15 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
+
+        // set up a NetworkStateListener
+        // This listener remains valid during the lifetime of this activity
+        NetworkStateListener networkStateListener = new NetworkStateListener();
+        IntentFilter intentFilter = new IntentFilter();
+        intentFilter.addAction("android.net.conn.CONNECTIVITY_CHANGE");
+        intentFilter.addAction("android.net.wifi.WIFI_STATE_CHANGED");
+
+        registerReceiver(networkStateListener, intentFilter);
     }
```

Comment:
register a receiver at login page. the receiver remains functioning until login page gets destoryed